tests/src/cosmwasm.spec.ts

```diff
@@ -315,3 +316,57 @@ test.serial("properly rollback first submessage if second fails", async (t) => {
   const gotNoFunds = await osmoClient.sign.getBalance(emptyAddr, osmosis.denomFee);
   t.is(gotNoFunds.amount, "0");
 });
+
+test.serial("query remote chain", async (t) => {
+  const { wasmClient, wasmController, link, osmoClient, osmoHost } = await demoSetup();
+
+  // there is an initial packet to relay for the whoami run
+  let info = await link.relayAll();
+  assertPacketsFromA(info, 1, true);
+
+  // get the account info
+  const accounts = await listAccounts(wasmClient, wasmController);
+  t.is(accounts.length, 1);
+  const { remote_addr: remoteAddr, channel_id: channelId } = accounts[0];
+  assert(remoteAddr);
+  assert(channelId);
+
+  // send some osmo to the remote address (using another funded account there)
+  const initFunds = { amount: "2500600", denom: osmosis.denomFee };
+  await osmoClient.sign.sendTokens(osmoClient.senderAddress, remoteAddr, [initFunds], "auto");
+
+  // make a new empty account on osmosis
+  const emptyAddr = randomAddress(osmosis.prefix);
+  const noFunds = await osmoClient.sign.getBalance(emptyAddr, osmosis.denomFee);
+  t.is(noFunds.amount, "0");
+
+  // from wasmd, send a packet to transfer funds from remoteAddr to emptyAddr
+  const sendFunds = { amount: "1200300", denom: osmosis.denomFee };
+  await remoteBankSend(wasmClient, wasmController, channelId, emptyAddr, [sendFunds]);
+
+  // relay this over
+  info = await link.relayAll();
+  assertPacketsFromA(info, 1, true);
+  // TODO: add helper for this
+  const contractData = parseAcknowledgementSuccess(info.acksFromB[0]);
+  // check we get { results : ['']} (one message with no data)
+  t.deepEqual(contractData, { results: [""] });
+
+  // ensure that the money was transfered
+  const gotFunds = await osmoClient.sign.getBalance(emptyAddr, osmosis.denomFee);
+  t.deepEqual(gotFunds, sendFunds);
+
+  // Use IBC queries to query account info from the remote contract
+  const ibcQuery = await wasmClient.sign.execute(
+    wasmClient.senderAddress,
+    wasmController,
+    {
+      ibc_query: {
+        channel_id: channelId,
+        msgs: [{ smart: { msg: toBase64(toUtf8(JSON.stringify({ list_accounts: {} }))), contract_addr: osmoHost } }],
```

Comment:
you can now use `toBinary` for those cases (where Binary referes to the CosmWasm Rust type):

```suggestion
        msgs: [{ smart: { msg: toBinary({ list_accounts: {} }), contract_addr: osmoHost } }],
```

with

```ts
import { toBinary } from "@cosmjs/cosmwasm-stargate";
```